lib/src/main/java/at/bitfire/cert4android/CustomCertStore.kt

```diff
@@ -0,0 +1,181 @@
+/***************************************************************************************************
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ **************************************************************************************************/
+
+package at.bitfire.cert4android
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.conscrypt.Conscrypt
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.security.KeyStore
+import java.security.Security
+import java.security.cert.CertificateException
+import java.security.cert.X509Certificate
+import java.util.logging.Level
+import javax.net.ssl.SSLContext
+
+class CustomCertStore private constructor(
+    private val context: Context,
```

Comment:
Is it really needed to store the context as a val? Since we are only using it here
```kotlin
val ui = UserDecisionUi.getInstance(context)
```
and `UserDecisionUi` is a Singleton, maybe we can just add it to the constructor, or initialize it lazily.